src/commands/infra/generate.ts

```diff
@@ -434,6 +437,7 @@ export const generateConfig = async (
         );
       }
     }
+    // TODO: handle definitionConfig === DefinitionYAMLExistence.LEAF_ONLY
```

Comment:
@NathanielRose @andrebriggs 

Revisiting this discussion... do we want to support scenarios where there is no `definition.yaml` and only leaf definitions?

Reply:
Once decision is made, this can be handled in a separate PR with a new user story. and i can work on it if needed

Reply:
Had a small conversation with @NathanielRose, and the consensus right now is opt out of having a "configuration C" and throw an error for the case of a missing parent `definition.yaml`. The reason is because `spk infra` is really designed around managing multi-cluster scenarios for Bedrock. Having individual infrastructure configuration would imply a _non_-multi-cluster setup, and in which case, the user should just handle the operations manually and not rely on `spk infra generate`.

Reply:
If it's not too much to ask for. Could you include this error handling in the code?

Reply:
@yradsmikham This sounds like a separate issue. @dennisseah @yradsmikham Let's a new issue for this bug/feature and address it in a new PR. Let's keep PR sizes small when possible

Reply:
@dennisseah @andrebriggs sure, I can create task for this, and work on it.

Reply:
@yradsmikham please share the new issue to groomed once you've created it

Reply:
@andrebriggs https://github.com/microsoft/bedrock/issues/918